File: arduinobot_test/src/simple_publisher.cpp
#include <ros/ros.h>
#include <std_msgs/String.h>
#include <sstream>

int main(int argc, char **argv)
{
    ros::init(argc, argv, "simple_publisher_cpp");
    ros::NodeHandle nh;

    auto pub = nh.advertise<std_msgs::String>("chatter", 10);
    auto rate = ros::Rate(10);

    int counter{};
    while (ros::ok())
    {
        std::stringstream ss;
        ss << "Hello world " << counter++;
        std_msgs::String message;
        message.data = ss.str();
        pub.publish(message);
        rate.sleep();
    }
}

File: arduinobot_test/scripts/simple_publisher.py
#!/usr/bin/env python3

import rospy
from std_msgs.msg import String


def main():
    rospy.init_node("simple_publisher_py", anonymous=True)
    pub = rospy.Publisher("chatter", String, queue_size=10)
    rate = rospy.Rate(10)
    counter: int = 0

    while not rospy.is_shutdown():
        pub.publish(f"Hello World {counter}")
        rate.sleep()
        counter += 1


if __name__ == "__main__":
    main()
